classes/controllers/grid/issues/IssueGridHandler.php

```diff
@@ -598,22 +599,50 @@ public function publishIssue($args, $request)
 
         // Send a notification to associated users if selected and context is publishing content online with OJS
         if ($request->getUserVar('sendIssueNotification') && $context->getData('publishingMode') != \APP\journal\Journal::PUBLISHING_MODE_NONE) {
+
+            // Notify users
             /** @var NotificationSubscriptionSettingsDAO $notificationSubscriptionSettingsDao */
             $notificationSubscriptionSettingsDao = DAORegistry::getDAO('NotificationSubscriptionSettingsDAO');
 
+            $userIdsToNotify = $notificationSubscriptionSettingsDao->getSubscribedUserIds(
+                [NotificationSubscriptionSettingsDAO::BLOCKED_NOTIFICATION_KEY],
+                [Notification::NOTIFICATION_TYPE_PUBLISHED_ISSUE],
+                [$contextId]
+            );
+
             $userIdsToMail = $notificationSubscriptionSettingsDao->getSubscribedUserIds(
-                [NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY],
+                [
+                    NotificationSubscriptionSettingsDAO::BLOCKED_NOTIFICATION_KEY,
+                    NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY
+                ],
                 [Notification::NOTIFICATION_TYPE_PUBLISHED_ISSUE],
                 [$contextId]
             );
 
+            [$userIdsToNotify, $userIdsToNotifyAndMail] = [
+                $userIdsToNotify->diff($userIdsToMail),
+                $userIdsToNotify->intersect($userIdsToMail)
+            ];
+
             $jobs = [];
-            foreach ($userIdsToMail->chunk(Mailer::BULK_EMAIL_SIZE_LIMIT) as $mailUserIds) {
+            foreach ($userIdsToNotify->chunk(PKPNotification::NOTIFICATION_CHUNK_SIZE_LIMIT) as $notifyUserIds) {
+                $notifyJob = new IssuePublishedMailUsers(
```

Comment:
Is this the right job? It looks the same as the mail job.

Reply:
Yes, it's should be one job but better to rename